src/Offer/CommandHandlers/DeleteCurrentOrganizerHandler.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Offer\CommandHandlers;
+
+use Broadway\CommandHandling\CommandHandler;
+use CultuurNet\UDB3\Offer\Commands\DeleteCurrentOrganizer;
+use CultuurNet\UDB3\Offer\OfferRepository;
+use CultuurNet\UDB3\UiTPAS\Validation\EventHasTicketSalesGuard;
+
+final class DeleteCurrentOrganizerHandler implements CommandHandler
+{
+    private OfferRepository $offerRepository;
+
+    private EventHasTicketSalesGuard $organizerChangeAllowedBasedOnTicketSales;
+
+    public function __construct(
+        OfferRepository $offerRepository,
+        EventHasTicketSalesGuard $organizerChangeAllowedBasedOnTicketSales
+    ) {
+        $this->offerRepository = $offerRepository;
+        $this->organizerChangeAllowedBasedOnTicketSales = $organizerChangeAllowedBasedOnTicketSales;
+    }
+
+    public function handle($command): void
+    {
+        if (!($command instanceof DeleteCurrentOrganizer)) {
+            return;
+        }
+
+        $offer = $this->offerRepository->load($command->getItemId());
```

Comment:
Are we sure this always gives an offer back? load() can throw an EventStreamNotFoundException, should we handle that here?

Reply:
Yes, if you include an unknown `id`, a new `Offer` is created, and you cannot delete an Organizer during a `create` only during an `update` so the `OfferRepository` should always return an `Offer`.
Whether this logic desirable is another matter, but I thought that was for legacy reasons like `OMD`.

Reply:
Throwing the exception `AggregateNotFoundException` is already handled inside the `ApiProblemFactory`. Inside the command handlers, there is no guarantee that the ID passed is that of an existing item.